### PR TITLE
Expose memmap dtype in data config

### PR DESCRIPTION
**Pull request by leon-g-xu**
This is expose dtype in the data config so that we can support reading memmap files with different dtypes

### CODE REVIEW

File: olmo/data/__init__.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ def build_memmap_dataset(
     return MemMapDataset(
         *paths,
         chunk_size=train_config.model.max_sequence_length,
+        memmap_dtype=train_config.data.effective_memmap_dtype,
```
[the remark this answers is not in the document]
Comment (leon-g-xu) [reply olmo/data/__init__.py]: Updated

File: olmo/config.py [hunk 2]
```diff
@@ -555,6 +556,7 @@ class InstanceFilterConfig(BaseConfig):
 @dataclass
 class DataConfig(BaseConfig):
     paths: Optional[List[str]] = None
+    memmap_dtype: Optional[str] = "uint16"
```
[the remark this answers is not in the document]
Comment (leon-g-xu) [reply olmo/config.py]: Updated

**Comment by leon-g-xu** [comment]
@2015aroras  Thanks for the review. Updated the PR to address the comment

**Review by 2015aroras — APPROVED** [review]
There are style issue due to imports. Make sure to follow the steps [here](https://github.com/allenai/OLMo/blob/9e894081e0c4632655ca879cd621357d9a26b7d8/.github/CONTRIBUTING.md#making-a-pull-request) so that required automatic checks pass.

**Comment by leon-g-xu** [comment]
@2015aroras Went though instructions and added all the missing things. Can I get another 👍  so that it kicks off all the auto checks?

**Comment by leon-g-xu** [comment]
> There are style issue due to imports. Make sure to follow the steps [here](https://github.com/allenai/OLMo/blob/9e894081e0c4632655ca879cd621357d9a26b7d8/.github/CONTRIBUTING.md#making-a-pull-request) so that required automatic checks pass.

@2015aroras  Went though instructions and added all the missing things. Can I get another 👍 so that it kicks off all the auto checks?

**Pull request merged**